credsweeper/file_handler/text_content_provider.py

```diff
@@ -23,5 +24,13 @@ def get_analysis_target(self) -> List[AnalysisTarget]:
             list of analysis targets based on every row in file
 
         """
-        lines = Util.read_file(self.file_path)
+        if Util.get_extension(self.file_path) == ".xml":
+            try:
+                tree = ElementTree.parse(self.file_path)
+                lines = Util.get_xml_data(tree.getroot())
+            except ElementTree.ParseError:
+                lines = Util.read_file(self.file_path)
+        else:
+            lines = Util.read_file(self.file_path)
```

Comment:
Duplicates...
Probably need to add an error_flag and split...

Reply:
Added `lines_init` flag [dbd77cd](https://github.com/Samsung/CredSweeper/pull/193/commits/dbd77cd6d36e6cf3457ed27018b881535471c40a).
Thank you for review.